src/main/java/util/QuerySplitter.java

```diff
@@ -0,0 +1,8 @@
+package util;
+
+public class QuerySplitter {
+
+    public static String[] queryWords(String query) {
```

Comment:
It doesn’t seem to me as something we need to keep as a util, it will look more readable if we keep it where it is used I think. @olegvovk96 wdyt?

Reply:
The same operation is used in 4-5 different places. Roman missed one part `trim()` when moving this method from the `SearchStrategy` interface to utility one. I would leave this one, but the method name `queryWords` should be changed because it doesn't say what it actually does.

```
default String[] queryWords(String query) {
   return query.trim().split("\\s+");
}
```
We can see that empty spaces are not handled

![image](https://user-images.githubusercontent.com/27965136/148745084-840ad107-05b6-4b4d-9e41-d706f5fd3715.png)

![image](https://user-images.githubusercontent.com/27965136/148745248-c3889e41-1cef-4840-80d1-24780e2b4c54.png)

Reply:
This class doesn't satisfy the requirements of the Utility class, please investigate and fix it.

Reply:
fixed

Reply:
Please pay attention on `private constructor` - https://rules.sonarsource.com/java/RSPEC-1118

Reply:
The names of the `class` and `method` are still unclear. In one place you work with `line` in another one with `query`. Why is it contain `Filter` in the name? It doesn't relate to filtering.

Reply:
class and method improved